internal/webhooks/clusterclass.go

```diff
@@ -274,27 +287,66 @@ func (webhook *ClusterClass) validateRemovedMachineDeploymentClassesAreNotUsed(c
 	return allErrs
 }
 
+func (webhook *ClusterClass) validateRemovedMachinePoolClassesAreNotUsed(clusters []clusterv1.Cluster, oldClusterClass, newClusterClass *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+
+	removedClasses := webhook.removedMachineClasses(oldClusterClass, newClusterClass)
+	// If no classes have been removed return early as no further checks are needed.
+	if len(removedClasses) == 0 {
+		return nil
+	}
+	// Error if any Cluster using the ClusterClass uses a MachinePoolClass that has been removed.
+	for _, c := range clusters {
+		for _, machinePoolTopology := range c.Spec.Topology.Workers.MachinePools {
+			if removedClasses.Has(machinePoolTopology.Class) {
+				// TODO: Same as above for MachineDeployments
+				allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "workers", "machinePools"),
+					fmt.Sprintf("MachinePoolClass %q cannot be deleted because it is used by Cluster %q",
+						machinePoolTopology.Class, c.Name),
+				))
+			}
+		}
+	}
+	return allErrs
+}
+
 func (webhook *ClusterClass) removedMachineClasses(oldClusterClass, newClusterClass *clusterv1.ClusterClass) sets.Set[string] {
```

Comment:
I think we need a separate func for MP. Otherwise we run into strange effects if we have MD & MP classes with the same names

Just duplicating it should be fine